### PR TITLE
Assignment 47 destroy

### PR DESCRIPTION
**Pull request by kenwmak**

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,12 +1,13 @@
 class CommentsController < ApplicationController
   def create
     @post = Post.find(params[:post_id])
+    @topic = Topic.find(@post.topic_id)
```
Comment (droberts84) [inline app/controllers/comments_controller.rb]: ActiveRecord associations provide a much better API then finding by `topic_id`.  You can just do `@post.topic

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
     @comment = Comment.new(post_id: params[:post_id], body: params[:comment][:body], user: current_user)
```
Comment (droberts84) [inline app/controllers/comments_controller.rb]: Nothing wrong with this, but using ActiveRecord associations, is there another way you can accomplish the something without having to specify the `post_id`?  Notice that you have already have `@post` initialized on line 3.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -15,15 +16,16 @@ def create
   
   def destroy
     @post = Post.find(params[:post_id])
+    @topic = Topic.find(@post.topic_id)
```
Comment (droberts84) [inline app/controllers/comments_controller.rb]: Same fix as bove

**Comment by kenwmak** [comment]
Good point! I changed those to use ```@topic = @post.topic_id``` instead

**Comment by droberts84** [comment]
That's very close, but topic_id gives you the id of the topic. You want the
actual topic instance. This is so common in Rails that ActiveRecord
associations will allow you to write @post.topic, thanks to the belongs_to
relationship

**Comment by kenwmak** [comment]
got it, thanks.

**Comment by droberts84** [comment]
What commit did you make the changes in?  I'm sure you got it but would appreciate you pushing it to this branch so I can double check everything looks good.

**Comment by kenwmak** [comment]
I submitted the change in this branch now https://github.com/kenwmak/bloccit/commit/b354cf0710debce2b91693730cca64e20bb49656

I was just making the changes on top of later checkpoint that i was working on, I will keep the changes in their respective branches

**Pull request merged**